### PR TITLE
Verify correct file size in sysio-write example

### PR DESCRIPTION
**Pull request by adammoody**
This adds a check to verify that the correct file size was created in the sysio-write example.  It should help catch bugs if the internal file size computation goes bad.

We can add similar tests to our travis suite after some write tests are available.

It also changes test_print() to always print a newline and call fflush, along with changing calls to this function to drop newlines.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the UnifyCR code style requirements.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted.

### CODE REVIEW

File: examples/src/sysio-write.c [hunk 1]
```diff
@@ -345,16 +345,32 @@ int main(int argc, char** argv)
...
+                test_print(rank, "%s size incorrect got %llu, expected %llu",
+                    targetfile, (unsigned long long)sbuf.st_size,
+                    (unsigned long long)expected_size);
+                ret = 1;
+            }
         } else {
-            test_print(rank, "stat(%s) failed: errno=%d (%s)",
+            /* our call to stat failed */
+            test_print(rank, "stat(%s) failed:",
                 targetfile, errno, strerror(errno));
```
Comment (CamStan) [inline examples/src/sysio-write.c]: @adammoody, looks like you might have dropped some format tags on line 371. There's one tag but three arguments on line 372.

Comment (dsikich) [reply examples/src/sysio-write.c]: yeah the "failed:" portion is missing an input

Comment (adammoody) [reply examples/src/sysio-write.c]: Nice catch!  Forgot to drop those arguments.  Turns out that test_print adds them if errno is non-zero, so dropped the formatting flags but forgot the args.

Comment (CamStan) [reply examples/src/sysio-write.c]: That works too!

**Comment by CamStan** [comment]
It building twice threw me for a sec. Realized you pushed a new branch.

**Pull request merged**